// @faustwp/core/dist/cjs/components/Toolbar/Toolbar.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Toolbar = void 0;
const react_1 = __importStar(require("react"));
const Edit_js_1 = require("./nodes/Edit.js");
const GraphiQL_js_1 = require("./nodes/GraphiQL.js");
const MyAccount_js_1 = require("./nodes/MyAccount.js");
const index_js_1 = require("../../wpHooks/index.js");
const ToolbarNode_js_1 = require("./ToolbarNode.js");
const useAuth_js_1 = require("../../hooks/useAuth.js");
const SiteName_js_1 = require("./nodes/SiteName.js");
/**
 * Renders a Toolbar that is based on WordPress' own toolbar.
 */
function Toolbar({ seedNode }) {
    /**
     * Define Toolbar Nodes that should be included by default.
     */
    const coreToolbarNodes = (0, react_1.useMemo)(() => {
        return [
            {
                id: 'site-name',
                location: 'primary',
                component: react_1.default.createElement(SiteName_js_1.SiteName, null),
            },
            {
                id: 'edit',
                location: 'primary',
                component: react_1.default.createElement(Edit_js_1.Edit, { seedNode: seedNode }),
            },
            {
                id: 'graphiql',
                location: 'primary',
                component: react_1.default.createElement(GraphiQL_js_1.GraphiQL, null),
            },
            {
                id: 'my-account',
                location: 'secondary',
                component: react_1.default.createElement(MyAccount_js_1.MyAccount, null),
                additionalClassNames: 'with-avatar',
            },
        ];
    }, [seedNode]);
    const [toolbarNodes, setToolbarNodes] = (0, react_1.useState)(coreToolbarNodes);
    const { isAuthenticated } = (0, useAuth_js_1.useAuth)();
    /**
     * Handle Toolbar nodes.
     */
    (0, react_1.useEffect)(() => {
        if (!isAuthenticated) {
            return;
        }
        const filteredNodes = index_js_1.hooks.applyFilters('toolbarNodes', coreToolbarNodes, {
            seedNode,
        });
        const uniqueIds = new Set(filteredNodes.map((nodes) => nodes.id));
        if (uniqueIds.size < filteredNodes.length) {
            throw new Error('Toolbar Nodes must have unique keys.');
        }
        setToolbarNodes(filteredNodes);
    }, [coreToolbarNodes, seedNode, isAuthenticated]);
    /**
     * Handle adding `admin-bar` body class on render.
     *
     * This could eventually be handled with a Faust bodyClass hook
     * @link https://developer.wordpress.org/reference/hooks/body_class/
     */
    (0, react_1.useEffect)(() => {
        if (!isAuthenticated) {
            return;
        }
        document === null || document === void 0 ? void 0 : document.body.classList.add('admin-bar');
        // Cleanup body class when this component unmounts.
        // eslint-disable-next-line consistent-return
        return () => {
            document === null || document === void 0 ? void 0 : document.body.classList.remove('admin-bar');
        };
    }, [isAuthenticated]);
    const primaryNodes = toolbarNodes.filter(({ location }) => location === 'primary');
    const secondaryNodes = toolbarNodes.filter(({ location }) => location === 'secondary');
    if (isAuthenticated !== true) {
        return null;
    }
    return (react_1.default.createElement("div", { id: "wpadminbar", className: "nojq" },
        react_1.default.createElement("div", { id: "wp-toolbar", className: "quicklinks", role: "navigation", "aria-label": "Toolbar" },
            react_1.default.createElement("ul", { id: "wp-admin-bar-root-default", className: "ab-top-menu" }, primaryNodes.map((_a) => {
                var { component, id } = _a, props = __rest(_a, ["component", "id"]);
                return (
                // eslint-disable-next-line react/jsx-props-no-spreading
                react_1.default.createElement(ToolbarNode_js_1.ToolbarNode, Object.assign({ key: id, id: id }, props), component));
            })),
            react_1.default.createElement("ul", { id: "wp-admin-bar-top-secondary", className: "ab-top-secondary ab-top-menu" }, secondaryNodes.map((_a) => {
                var { component, id } = _a, props = __rest(_a, ["component", "id"]);
                return (
                // eslint-disable-next-line react/jsx-props-no-spreading
                react_1.default.createElement(ToolbarNode_js_1.ToolbarNode, Object.assign({ key: id, id: id }, props), component));
            })))));
}
exports.Toolbar = Toolbar;
